python/cudf/cudf/tests/test_series.py

```diff
@@ -630,3 +630,26 @@ def test_series_mode(df, dropna):
     actual = df.mode(dropna=dropna)
 
     assert_eq(expected, actual, check_dtype=False)
+
+
+@pytest.mark.parametrize(
+    "sr1", [pd.Series([10, 11, 12], index=["a", "b", "z"]), pd.Series(["a"])]
+)
+@pytest.mark.parametrize(
+    "sr2", [pd.Series([]), pd.Series(["a", "a", "c", "z", "A"])]
+)
+@pytest.mark.parametrize(
+    "op",
+    ["__eq__", "__ne__", "__lt__", "__gt__", "__le__", "__gt__", "__ge__"],
+)
+def test_series_error_equality(sr1, sr2, op):
+    gsr1 = cudf.from_pandas(sr1)
+    gsr2 = cudf.from_pandas(sr2)
+
+    try:
+        sr1.getattr(op)(sr2)
+    except Exception as e:
+        with pytest.raises(type(e), match=e.__str__()):
+            gsr1.getattr(op)(gsr2)
+    else:
+        raise AssertionError(f"Expected {op} to fail for {sr1} and {sr2}")
```

Comment:
This is a really nice pattern! I wonder if there's a way to encapsulate it so it can be reused across multiple tests.

Reply:
I build a common utility(`assert_exception`) around this pattern, this function takes two callables and compares their exceptions.

This pattern is widely used in our pytests, so I'd like to replace all those occurrences aswell. So would you be okay with this new addition?

Also, pep-8 [complains](https://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def)  about creating `lambda` functions(instead suggests to use `def`) which are more readable in cases like this, hence should we silence `E731` errors for the whole project or do it specifically in case-by-case basis in pytests?

cc: @kkraus14

Reply:
+1 to replacing this pattern elsewhere in a follow up PR. Added a comment suggestion below for handling the `lambda` issues

Reply:
Awesome! Suggest a slightly more descriptive name like `assert_exceptions_equal`

Reply:
Renamed the function and added the ability to pass function arguments so we don't have to create and pass `lambda` functions. This way we don't have flake8 complaining of a `pep-8` violation